test/Integrations/Insightly/InsightlyClientTest.php

```diff
@@ -126,6 +126,11 @@ public function it_can_manage_opportunities(): void
 
         $this->opportunityId = $this->insightlyClient->opportunityResource()->create($expectedOpportunity);
 
+        // When a create is done on Insightly not all objects are stored immediately
+        // When getting the created object it can happen some parts like linked contact and custom fields are still missing
+        // This sleep will fix that 😬
+        sleep(1);
```

Comment:
Nit: You could of course also try the `get` a specific amount of times until you get the correct response or mark the test as failed if you didn't get the expected response after x tries. But maybe we don't need to overthink this for now.

Reply:
Indeed, that's always better then a static sleep. I created a follow-up ticket to fix some already introduced Technical Debt: https://jira.uitdatabank.be/browse/PROJ-187